pkgs/dart_tooling_mcp_server/test/tools/dtd_test.dart

```diff
@@ -279,6 +221,173 @@ void main() {
           );
         });
       });
+
+      group('runtime errors', () {
+        final errorCountRegex = RegExp(r'Found \d+ errors?:');
+
+        setUp(() async {
+          await testHarness.startDebugSession(
+            counterAppPath,
+            'lib/main.dart',
+            isFlutter: true,
+            args: ['--dart-define=include_layout_error=true'],
+          );
+        });
+
+        test('can be read and cleared using the tool', () async {
+          final tools =
+              (await testHarness.mcpServerConnection.listTools()).tools;
+          final runtimeErrorsTool = tools.singleWhere(
+            (t) => t.name == DartToolingDaemonSupport.getRuntimeErrorsTool.name,
+          );
+          late CallToolResult runtimeErrorsResult;
+
+          // Give the errors at most a second to come through.
+          var count = 0;
+          while (true) {
+            runtimeErrorsResult = await testHarness.callToolWithRetry(
+              CallToolRequest(
+                name: runtimeErrorsTool.name,
+                arguments: {'clearRuntimeErrors': true},
+              ),
+            );
+            expect(runtimeErrorsResult.isError, isNot(true));
+            final firstText =
+                (runtimeErrorsResult.content.first as TextContent).text;
+            if (errorCountRegex.hasMatch(firstText)) {
+              break;
+            } else if (++count > 10) {
+              fail('No errors found, expected at least one');
+            } else {
+              await Future<void>.delayed(const Duration(milliseconds: 100));
+            }
+          }
+          expect(
+            (runtimeErrorsResult.content[1] as TextContent).text,
+            contains('A RenderFlex overflowed by'),
+          );
+
+          final nextResult = await testHarness.callToolWithRetry(
+            CallToolRequest(name: runtimeErrorsTool.name),
+          );
+          expect(
+            (nextResult.content.first as TextContent).text,
+            contains('No runtime errors found'),
```

Comment:
I'm surprised there are no errors even though the clearRuntimeErrors arg was not passed, but maybe I'm not following the logic here.

Reply:
The logic is a bit hard to follow - we are passing `clearRuntimeErrors` in the retry loop above (which is waiting for the error to show up, this test was flaking without that since there is no explicit 1 second delay any more built into this tool).

This then asks for the errors again - to check that the errors were in fact cleared.

Reply:
Added a comment above this tool call